### DISCUSSION TITLE
Access to InnerJob of MicrosoftDependencyInjectionJobFactory.ScopedJob not possible

### DISCUSSION THREAD

**Discussion by nstdspace** [quartznet/quartznet#1578] [Q&A]
Edit: @lahma, Thank you for moving this to the correct place!

I want to use `MicrosoftdependencyInjectionJobFactory `to enable dependency injection for job classes. The Problem is, I have some logic in an `IJobListener` which relies on the type of the job instance (and has to call methods on it). Since the wrapper class `ScopedJob` is private, I cannot access the `InnerJob` property, which I need in the listener. 

Is there any particular reason why the wrapper class is private? Is there a better solution than using reflection to access the private field or replicating the job factory myself?

Thank you!

**Comment by lahma** [comment]
Well generally I prefer the approach of keeping everything private/internal until needed otherwise. Gives a lot more jiggle room as there's no breaking changes to the public API.

If you mean that `IJobExecutionContext.JobInstance` is of wrapper type, it might make sense to unwrap as the actual job that is behind the `ScopedJob`.

**Reply by nstdspace** [reply]
Thank you for the fast response! Yes, this is exactly what I mean. 
How could I 'unwrap' it?

**Reply by lahma** [reply]
I think the ideal thing would be to have this on Quartz's side `internal interface IJobWrapper { IJob Target { get; } }` that `ScopedJob` would implement,  which would be unwrapped in job execution context. So currently no out of the box solution (a PR would be welcome!). 

So you now have to result to using reflection (and create a test case against it so it doesn't break in version upgrades).

**Reply by nstdspace** [reply]
Okay, I misunderstood you first. I already have a solution using reflection.
I like the idea with the JobWrapper interface - if I find time for this I can make a PR.
